reference-manual/src/main/java/migration/Migration.java

```diff
@@ -0,0 +1,192 @@
+        FunctionEx<Map.Entry<String, Integer>, EntryProcessor<String, Integer, Void>> entryProcFn =
+                entry ->
+                        (EntryProcessor<String, Integer, Void>) e -> {
+                            e.setValue(e.getValue() == null ? 1 : e.getValue() + 1);
+                            return null;
+                        };
+        Sinks.mapWithEntryProcessor(map, Map.Entry::getKey, entryProcFn);
+
+        /*
+        FunctionEx<Map.Entry<String, Integer>, EntryProcessor<String, Integer>> entryProcFn =
+                entry ->
+                        (EntryProcessor<String, Integer>) e -> {
+                            e.setValue(e.getValue() == null ? 1 : e.getValue() + 1);
+                            return null;
+                        };
+        Sinks.mapWithEntryProcessor(map, Map.Entry::getKey, entryProcFn);
+        */
+        //end::entryProcessor1[]
+    }
+
+    static void serviceFactory() {
+        //tag::serviceFactory1[]
+        ServiceFactories.sharedService(ctx -> Executors.newFixedThreadPool(8), ExecutorService::shutdown);
+        //ContextFactory.withCreateFn(jet -> Executors.newFixedThreadPool(8)).withLocalSharing();
+
+        ServiceFactories.nonSharedService(ctx -> DateTimeFormatter.ofPattern("HH:mm:ss.SSS"), ConsumerEx.noop());
+        //ContextFactory.withCreateFn(jet -> DateTimeFormatter.ofPattern("HH:mm:ss.SSS"))
+        //end::serviceFactory1[]
+
+        Pipeline pipeline = Pipeline.create();
+        BatchStage<Runnable> stage = pipeline.readFrom(TestSources.items(1, 2, 3)).map(i -> (Runnable) () -> {});
+
+        //tag::serviceFactory2[]
+        stage.mapUsingServiceAsync(
+                ServiceFactories.sharedService(ctx -> Executors.newFixedThreadPool(8)),
+                2,
+                false,
+                (exec, task) -> CompletableFuture.supplyAsync(() -> task, exec)
+        );
+
+        /*
+        stage.mapUsingContextAsync(
+                ContextFactory.withCreateFn(jet -> Executors.newFixedThreadPool(8))
+                        .withMaxPendingCallsPerProcessor(2)
+                        .withUnorderedAsyncResponses(),
+                (exec, task) -> CompletableFuture.supplyAsync(() -> task, exec)
+        );
+        */
+        //end::serviceFactory2[]
+    }
+
+    static void async() {
+        Pipeline p = Pipeline.create();
+        ServiceFactory<?, ExecutorService> serviceFactory = ServiceFactories.sharedService(ctx -> Executors.newFixedThreadPool(8));
+
+        StreamStage<Long> stage = p.readFrom(TestSources.itemStream(3))
+                .withoutTimestamps()
+                .map(SimpleEvent::sequence);
+
+        //tag::async1[]
+        stage.mapUsingServiceAsync(serviceFactory,
+                (executor, item) -> {
+                    CompletableFuture<Traverser<String>> f = new CompletableFuture<>();
+                    executor.submit(() -> f.complete(traverseItems(item + "-1", item + "-2", item + "-3")));
```

Comment:
it would be better to return a List<> and then use Traversers.traverseIterable() in the following step